tests/test_downloader_handlers.py

```diff
@@ -420,6 +420,26 @@ def _clean(data):
         if errback:
             deferred.addErrback(errback)
         return deferred
+    
+    def test_ftp_list_success(self):
+        request = Request(url="ftp://127.0.0.1:%s/" % self.portNum,
```

Comment:
What about `ftp://127.0.0.1:%s` urls (without a trailing slash)? I think it'd be nice to show listing for them as well.